engine/runtime-compiler/src/main/scala/org/enso/compiler/context/LocalScope.scala

```diff
@@ -34,11 +35,27 @@ class LocalScope(
   final val aliasingGraph: () => AliasGraph,
   final private val scopeProvider: () => AliasGraph.Scope,
   final private val dataflowInfoProvider: () => DataflowAnalysis.Metadata,
-  final val flattenToParent: Boolean                       = false,
-  private val parentFrameSlotIdxs: Map[AliasGraph.Id, Int] = Map()
+  final private val symbolsProvider: () => FramePointerAnalysis.FramePointerMeta =
+    null,
+  final val flattenToParent: Boolean                             = false,
+  private val parentFrameSlotIdxs: () => Map[AliasGraph.Id, Int] = () => Map()
 ) {
   lazy val scope: AliasGraph.Scope                 = scopeProvider()
   lazy val dataflowInfo: DataflowAnalysis.Metadata = dataflowInfoProvider()
+  def allSymbols(where: String): List[String] = {
+    def symbols() = scope.allDefinitions.map(_.symbol)
+    val meta      = if (symbolsProvider == null) null else symbolsProvider()
+    if (meta != null && meta.variableNames.isDefined) {
+      val cached = meta.variableNames.get
+      FramePointer.assertSame(where, cached, () => symbols())
+    } else {
+      val result = symbols()
+      System.err.println(
+        "BAD!!!! Computing from scope at " + where + " = " + result
+      )
```

Comment:
Let's replace this with `Logger` and use it in a test to detect no `BAD!!!` situations happen. Btw. they do happen when using `EvalNode` and I believe that's necessary as it evaluates random code and needs the `LocalScope` with `AliasGraph`.

Reply:
Done in c87f67a0b65c926cf1b3c880150f4576b7a91475 - use:
```
./enso --log-level trace --run xyz.enso | grep "] Scope"
```
to see messages about _"Scope computed from AliasAnalysis"_.